iceoryx_hoofs/include/iceoryx_hoofs/concurrent/lockfree_queue.hpp

```diff
@@ -102,7 +102,6 @@ class LockFreeQueue
 
   protected:
     using Queue = IndexQueue<Capacity>;
-    using BufferIndex = typename Queue::value_t;
```

Comment:
I can revert the changes here but I removed the type alias because it was actually just an alias for `uint64_t`, the default `ValueType` of the `IndexQueue`.

Reply:
It would be better to alias it to avoid complications should `typename Queue::value_t` change for some reason (and compatible in principle with `uin64_t` but produce bugs for some weird reason, such as relying on overflow of an `uint32_t`).

Reply:
I tend to keep the current changes since the index type was not used so far and it makes testing less complex.

Reply:
I agree with keeping the current changes.